tests/test_misc.py

```diff
@@ -1699,170 +1692,217 @@ def test_filter_variables_pool_exclude_regex_patterns(
+    mock_output_manager._route_save_functions(
+        "json_file",
+        "save_path",
+        {"key": {"var": "value"}},
+        True,
+        {"filters": "regex"},
+        "graphics_dir",
+    )
+    mock_output_manager._dict_to_file_json.assert_called_once_with(
+        {"key": {"var": "value"}}, os.path.join("save_path", "filename.json")
     )
+    # Restore original method
+    mock_output_manager._dict_to_file_json = output_manager_original_method_states[
+        "_dict_to_file_json"
+    ]
+    mock_output_manager._route_save_functions = output_manager_original_method_states[
+        "_route_save_functions"
+    ]
+
+
+def test_route_save_functions_graph(
+    mock_output_manager: OutputManager,
+    output_manager_original_method_states: Dict[str, Callable],
+) -> None:
     with patch(
         "RUFAS.graph_generator.GraphGenerator.generate_graph"
     ) as mock_generate_graph:
+        mock_output_manager.add_warning = MagicMock()
         graph_data = {"filters": ".*", "other keys": "other values"}
-        mock_output_manager._load_filter_file_content = MagicMock(
-            return_value=graph_data
-        )
-        mock_output_manager.save_variables(
-            "dummy_path",
-            "dummy_dir_path/",
-            produce_graphics=False,
-            graphics_dir=Path("graphics"),
+        mock_output_manager._route_save_functions(
+            "graph_file",
+            "save_path",
+            {"key": {"var": "value"}},
+            False,
+            graph_data,
+            "graphics_dir",
         )
         mock_generate_graph.assert_not_called()
-
-        graph_data = {"no_filters": ".*", "other keys": "other values"}
-        mock_output_manager._load_filter_file_content = MagicMock(
-            return_value=graph_data
-        )
-        mock_output_manager.save_variables(
-            "dummy_path",
-            "dummy_dir_path/",
-            produce_graphics=True,
-            graphics_dir=Path("graphics"),
+        mock_output_manager.add_warning.assert_called_once_with(
+            "No Graphics",
+            "Graphic generation is disabled, skipping filter_file='graph_file'",
+            {"class": "OutputManager", "function": "_route_save_functions"},
         )
-        mock_generate_graph.assert_not_called()
 
-        graph_data = {"filters": ".*", "other keys": "other values"}
-        mock_output_manager._load_filter_file_content = MagicMock(
-            return_value=graph_data
-        )
-        mock_output_manager.save_variables(
-            "dummy_path",
-            "dummy_dir_path/",
-            produce_graphics=True,
-            graphics_dir=Path("graphics"),
+        mock_output_manager._route_save_functions(
+            "graph_file",
+            "save_path",
+            {"key": {"var": "value"}},
+            True,
+            graph_data,
+            "graphics_dir",
         )
         mock_generate_graph.assert_called_once_with(
```

Comment:
```suggestion
        )
        assert mock_output_manager.add_warning.call_count == 1
        mock_generate_graph.assert_called_once_with(
```
This would ensure that `add_warning` is not somehow being called when it shouldn't be.